sources/models/src/lib.rs

```diff
@@ -237,6 +237,11 @@ struct ECSSettings {
     loglevel: ECSAgentLogLevel,
     enable_spot_instance_draining: bool,
     image_pull_behavior: ECSAgentImagePullBehavior,
+    container_stop_timeout: ECSDurationValue,
+    task_cleanup_wait_duration: ECSDurationValue,
+    metadata_service_steady_state_rate: i64,
+    metadata_service_burst_rate: i64,
```

Comment:
> I am actually fine with accepting negative values here if ecs-agent also tolerates them, which it appears to. What numeric type does it use? It's clearly a signed type. Can we just use that?

When the ECS agent loads configurations, it'll throw a warning if it finds negative values for either `TaskMetadataSteadyStateRate` or `TaskMetadataBurstRate`. Instead, [it'll use the defaults.](https://github.com/aws/amazon-ecs-agent/blob/5fb8e801e9d11630c3470abb778291913ebb9d7f/agent/config/config.go#L360-L364)

So yes, negative numbers are "valid" values in the sense that they can be deserialized into the Go struct, but the program logic will reject them.

> (Note that if it's actually an i64, then using a u64 here would be an error that could expose the agent to invalid input.)

Not sure if this is relevant, but [the ecs-agent defines both these values as a Go `int`](https://github.com/aws/amazon-ecs-agent/blob/5fb8e801e9d11630c3470abb778291913ebb9d7f/agent/config/types.go#L271-L275) which is signed and should be 64 bits wide on a x64 system (and 32 on a x32 system). So yes, a `u64` integer could overflow a signed `int` providing an invalid value

Reply:
> I am pretty unhappy with this rationale and frankly just tired of accepting all the recent justifications along these lines. (See also https://github.com/bottlerocket-os/bottlerocket/issues/2463, https://github.com/bottlerocket-os/bottlerocket/issues/2257.)

I think I can fix #2257, I added a comment there.

Reply:
> Especially for the datastore serializer, i64 seems like it would have the same problems as u64 would for JSON. Is it actually problematic or does serde_json do the right thing in both cases?

The serializer only complains about `u64`:

```bash
bash-5.1# apiclient set ecs.metadata-service-burst-rate=1
Failed to change settings: Failed PATCH request to '/settings?tx=apiclient-set-cFwvZoTEUPHub2iq': Status 500 when PATCHing /settings?tx=apiclient-set-cFwvZoTEUPHub2iq: Error serializing Settings: 'u64' not allowed by Serializer
```

`i64` does work fine.